### ISSUE TITLE
Plugin migration files not being refreshed in correct order.

### ISSUE THREAD

**Issue by justin-lau** [octobercms/october#1356]
The change in #1304 makes plugin migration files not refreshing in the correct (reversed) order with the `plugin:refresh` command.

**Reproducing:**
1. Create a plugin with 2 migration files that have models with a one to many relationship.
2. Make sure they are listed in correct order in `version.yaml`.
3. Run `php artisan plugin:refresh` for the first time and it should migrate properly.
4. Run `php artisan plugin:refresh` again. It should throw a foreign key violation SQL exception.

Commenting out the PR's code does resolve the issue.

**Comment by lazycms** [comment]
兄弟 看英文老费劲了

**Comment by gabsource** [comment]
Can not reproduce your issue with the step suggested. __The rollback succeed even with #1304__.

Test files : 

```yaml
1.0.1:
 - First
 - create_first_table.php
1.0.2:
 - Second
 - create_second_table.php
```

```PHP
<?php namespace RainLab\User\Updates;

use Schema;
use October\Rain\Database\Updates\Migration;

class CreateFirstTable extends Migration
{

    public function up()
    {
        Schema::create('fix_demo_table_1', function($table)
        {
            $table->engine = 'InnoDB';
            $table->increments('id');
            $table->timestamps();
        });
        echo '1';
    }

    public function down()
    {
        Schema::dropIfExists('fix_demo_table_1');
    }

}

```

```PHP
<?php namespace RainLab\User\Updates;

use Schema;
use October\Rain\Database\Updates\Migration;

class CreateSecondTable extends Migration
{

    public function up()
    {
        Schema::create('fix_demo_table_2', function ($table) {
            $table->engine = 'InnoDB';
            $table->integer('fix_id')->unsigned();
            $table->foreign('fix_id')->references('id')->on('fix_demo_table_1')->onDelete('restrict');
        });
        echo '2';
    }


    public function down()
    {
        Schema::dropIfExists('fix_demo_table_2');
    }

}

```

```
Rolled back: Fix.Version
Reinstalling plugin...
Fix.Version
 - v1.0.1:  First
 - v1.0.2:  Second
```

**Comment by justin-lau** [comment]
@gabsource try using both migration files in a same version, and you should see the exception:

```yaml
1.0.1:
 - First
 - create_first_table.php
 - create_second_table.php
```

**Comment by gabsource** [comment]
Confirmed, the sorting from PR #1304 was inconsistent. Patch proposed in PR #1366.

**Comment by daftspunk** [comment]
Fixed by #1366
